phoenix-core/src/it/java/org/apache/phoenix/end2end/QueryLoggerIT.java

```diff
@@ -250,7 +250,44 @@ public void testWithLoggingOFF() throws Exception{
         assertFalse(foundQueryLog);
         conn.close();
     }
-    
+
+    @Test
+    //DROP Table statement should be always logged.
+    public void testDropTableStatement() throws Exception{
+        String tableName = generateUniqueName();
+        createTableAndInsertValues(tableName, true);
+        Properties props= new Properties();
+        props.setProperty(QueryServices.LOG_LEVEL, LogLevel.INFO.name());
+        Connection conn = DriverManager.getConnection(getUrl(),props);
+        assertEquals(conn.unwrap(PhoenixConnection.class).getLogLevel(),LogLevel.INFO);
+
+        String query = "SELECT * FROM " + tableName;
+        ResultSet rs = conn.createStatement().executeQuery(query);
+
+        String query2 = "DROP TABLE " + tableName;
+        conn.createStatement().execute(query2);
+        while (rs.next()) {
+            rs.getString(1);
+            rs.getString(2);
+        }
+        String logQuery = "SELECT * FROM " + SYSTEM_CATALOG_SCHEMA + ".\"" + SYSTEM_LOG_TABLE
+            + "\"";
+        int delay = 5000;
+
+        // sleep for sometime to let query log committed
+        Thread.sleep(delay);
```

Comment:
Instead of just waiting once and checking, use a retry logic for 10 times and wait 1 sec for each iteration. This will make tests more reliable and possibly faster (if it syncs fast)

Reply:
Changed.